### PR TITLE
Fix test performance issue on CentOS (#4085)

### LINKED ISSUE #4085: Perl perf issue in =~ (binding operator) on verilated.cpp
An off-topic (not Verilator issue) - on regression runs in test_regress.

On a CentOS I see t_dist_cppstyle.pl hangs for hours together. With some prints I located the issue to be around "binding operator" in Perl - for some big (not really big) file such as verilated.cpp

```perl
  printf ( "$filename $message $pattern %0d \n", length($buffer));
    while ($buffer =~ s/.*?^($pattern)//sm) {
      print "AsFigo2: $filename $message \n";
```
I got below:
```log
 ../include/verilated.cpp 'virtual' keyword is redundant on 'override' method (?^:[^/]*virtual[^{};]+override) 130126 
```
I don't see the 2nd printf inside that while-loop executing at all (even after 4-5 hours). Sure, this laptop is not the top-of-the-line config/fully loaded, but it does run quite a few EDA tools reasonably well. 

For experimentation sake, I emptied verilated.cpp and the issue shows up on next file. 

Any clues on how we can fix it cleanly please?

### PR DESCRIPTION
**Pull request by svenka3**
Fixes #4085 

Looking at 3 lines at a time instead of entire file content.

### CODE REVIEW

File: test_regress/t/t_dist_cppstyle.pl [hunk 1]
```diff
@@ -63,18 +63,30 @@ sub checkPattern {
     my $pattern = shift;
     my $message = shift;
 
-    my $offset = 0;
-    my $buffer = $contents;
-    while ($buffer =~ s/.*?^($pattern)//sm) {
-      my $lineno = offset_to_lineno($contents, $offset + $-[-1]);
-      $offset += $+[1];
-      error("$filename:$lineno: $message");
+    open(my $in_fileh, $filename)
```
Comment (wsnyder) [inline test_regress/t/t_dist_cppstyle.pl:66]:
```suggestion
    my $in_fileh = IO::File-new("<$filename")
```

File: test_regress/t/t_dist_cppstyle.pl [hunk 2]
```diff
@@ -63,18 +63,30 @@ sub checkPattern {
...
+    my $lineno = 0;
+    my $start_row = 0;
+    my $end_row = 0;
+    my $buffer;
+
+    my $num_rows = scalar(@lines_l);
+    my @n_rows;
+
+    for( my $row_i = 0; $row_i < $num_rows; $row_i++) {
+        $lineno++;
```
Comment (wsnyder) [inline test_regress/t/t_dist_cppstyle.pl:78]:
```suggestion
        $++lineno;
```

File: test_regress/t/t_dist_cppstyle.pl [hunk 3]
```diff
@@ -63,18 +63,30 @@ sub checkPattern {
...
+        $start_row = $row_i;
+        $end_row = $start_row + 3;
```
Comment (wsnyder) [inline test_regress/t/t_dist_cppstyle.pl:80]:
```suggestion
        my $start_row = $row_i;
        my  $end_row = $start_row + 3;
```

File: test_regress/t/t_dist_cppstyle.pl [hunk 4]
```diff
@@ -63,18 +63,30 @@ sub checkPattern {
...
+        if ($end_row >= $num_rows) {
+            $end_row = $num_rows - 1;
+        }
+        @n_rows = @lines_l [$start_row..$end_row];
+        $buffer = join ("", @n_rows);
```
Comment (wsnyder) [inline test_regress/t/t_dist_cppstyle.pl:85]:
```suggestion
        my $buffer = join ("", @lines_l [$start_row..$end_row]);
```

Comment (svenka3) [reply test_regress/t/t_dist_cppstyle.pl:85]: Thanks for the comments - is there a simple way to "accept all changes"?

**Comment by wsnyder** [comment]
Not as far as I know.

File: test_regress/t/t_dist_cppstyle.pl (hunk 2 above)
Comment (svenka3) [reply test_regress/t/t_dist_cppstyle.pl:78]: Did you mean ++$lineno;

File: test_regress/t/t_dist_cppstyle.pl (hunk 3 above)
Comment (svenka3) [reply test_regress/t/t_dist_cppstyle.pl:80]: This suggestion of "my " in that same line (as expression) - would this mean lines 70-71 are redundant? Should I remove those?

```perl
  my $start_row = 0;
    my $end_row = 0;
```

Comment (wsnyder) [reply test_regress/t/t_dist_cppstyle.pl:80]: Oh, correct.

File: test_regress/t/t_dist_cppstyle.pl [hunk 5]
```diff
@@ -63,18 +63,29 @@ sub checkPattern {
...
+        my $start_row = $row_i;
+        my $end_row = $start_row + 3;
+        if ($end_row >= $num_rows) {
+            $end_row = $num_rows - 1;
+        }
+        @n_rows = @lines_l [$start_row..$end_row];
+        my $buffer = join ("", @n_rows);
+        if ($buffer =~ s/.*?^($pattern)//sm) {
+            error("$filename:$lineno: $message");
+        }
```
Comment (wsnyder) [inline test_regress/t/t_dist_cppstyle.pl:88]:
```suggestion
    my $lineno = 0;
    my $num_rows = scalar(@lines_l);
    for (my $row_i = 0; $row_i < $num_rows; ++$row_i) {
        ++$lineno;
        my $start_row = $row_i;
        my $end_row = $start_row + 3;
        if ($end_row >= $num_rows) {
            $end_row = $num_rows - 1;
        }
        my $buffer = join("", @lines_l[$start_row..$end_row]);
        if ($buffer =~ s/.*?^($pattern)//sm) {
            error("$filename:$lineno: $message");
        }
```

**Review by wsnyder — COMMENTED** [review]
Missed some first round, with this should be good.

**Comment by wsnyder** [comment]
Merging a similar patch manually shortly.

**Pull request closed without merging**